aerogear/voyager-ionic-example
# e2e subscriptions

first pass at a full e2e working subscriptions example. 

Note: there is some technical debt in this PR that could potentially be resolved upstream. Currently apollo does not provide very good support for subscriptions and optimistic responses. This means that without some form of deduplication any tasks created on a client receive both the subscription and the optimistic UI and display both. Currently working around this using lodash. Definitely not ideal but will update here once I have looked upstream.

First pass at a full working subscriptions implementation using the example server.

Verification:

1. Run the example server
1. Run the application
1. Open a second (incognito) window to create a second client
1. Create a task - you should see this new task (without duplicates) on both
1. Update a task - you should see this on both
1. Delete a task - you should see this on both
1. Go offline and perform some CUD actions as above
1. When you return online your changes should be synced to both clients
